plugins/catalog-backend/src/search/Collator.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  IndexableDocument,
+  IndexableDocumentCollator,
+} from '@backstage/plugin-search-indexer-backend';
+import { EntitiesCatalog } from '../catalog';
+
+export interface CatalogEntityDocument extends IndexableDocument {
+  componentType: string;
+}
+
+export const SearchCollatorFactory = (
```

Comment:
Yeah, in my head originally, there wasn't really a need for a factory wrapper. You'd just have...

1. A particular plugin implementing a method that returns all of the things it wants indexed.
2. A process that calls all of the registered methods from all of the relevant plugins and then dumps those things into a search engine index

But it turns out you can't just pull a catalog-aware thing out of thin air, you need some kind of pre-configured client injected, and and instance of such a client seems to live in `packages/backend`.  Likewise, zooming out from catalog, TechDocs would need its own dependency injected thing (that doesn't exist yet) that is aware of every TechDocs site, a configured instance of which would probably live in `packages/backend`.  And so on.

I think in an ideal world, if `catalog-backend` and the search indexer were installed as peers in a backstage backend instance, you wouldn't actually have to do anything in code for catalog entities to become indexed.  It would just happen according to your configurations.

But given how an instance of backstage backend works with respect to backend plugins, that doesn't seem feasible at the moment.  On the other hand, moving all of this chunk of code to `packages/backend` seems like a swing too far in the other direction: "So you want to get search up and running? Okay, don't forget to implement your own collator for every type of thing you want indexed."

Which is how we landed on this bit of code here.

Given that context, what would you recommend?